src/tooling/grain_doc.md

```diff
@@ -0,0 +1,152 @@
+ */
+export let get = (index, array) => {
+  array[index]
+}
+```
+
+## Usage
+
+Before running `grain doc`, you'll want to add docblock comments to your code. Docblock comments are started with `/**` and ended with `*/`. They must exist on the line above a type declaration or exported value. No documentation will be generated for non-exported values—to keep module internals private.
+
+You can run `grain doc <file|dir> -o <file|dir>` to generate markdown documentation for your code. As of Grain v0.5, the `grain doc` command supports directory input to recursively generate documentation for all `.gr` files.
+
+### Without docblocks
+
+You can even use `grain doc` without docblocks in your code. It will still generate simple documentation for data types and exported values, such as displaying their names and type signatures.
+
+## Attributes
+
+### @param
+
+```gr
+/**
+ * @param name: description
+ */
+```
+The `@param` attribute provides information for one parameter of the function being documented.
+
+Multiple `@param` attributes should be specified in order of the function parameters. The name specified before the colon (`:`) should match the parameter name exactly—while not currently enforced, it will be required for named parameters in the future. The content after the colon is the description of the parameter.
+
+When generating docs, the type of the parameter will be detected by the compiler and added to the parameter's row in the "Parameters" table.
+
+### @returns
+
+```gr
+/**
+ * @returns description
+ */
+```
+
+The `@returns` attribute provides information for the return value of the function being documented.
+
+This attribute requires a description of the return value. In the Grain standard library, we don't use the `@returns` attribute if the function returns `Void`.
+
+When generating docs, the type of the return value will be detected by the compiler and added to the row in the "Returns" table.
+
+### @example
+
+```gr
+/**
+ * @example code
+ */
+```
+
+The `@example` attribute provides snippets showing how the documented code is used. Currently, only single line examples are allowed.
+
+When generating docs, the code snippet will be wrapped in a code block tagged for the `grain` language and added to an "Examples" section.
+
+### @since
+
+```gr
+/**
+ * @since version
+ */
+```
+
+The `@since` attribute provides the version in which the documented code was added.
+
+This attribute requires a [Semantic Version](https://semver.org/), optionally prefixed with `v`. For example,`v0.2.0` or `0.2.0` are treated the same.
+
+When generating docs, a `<details>` element titled "Added in X.X.X" will be added above the type signature. When this attribute is specified, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "Added in next", instead of the version.
```

Comment:
```suggestion
When generating docs, a `<details>` element titled "Added in X.X.X" will be added above the type signature. When this attribute is specified, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify unreleased versions in your source code, and the generated docs will show "Added in next", instead of the version.
```